### PR TITLE
Refactor buttons

### PR DESCRIPTION
**Pull request by ptbrowne**
https://ptbrowne.github.io/cozy-ui/react/

@gregorylegarec in #443 needs to have easy access to button themes.
In this refactoring, I tried to make more visible the fact that button variants all share the same code and are based on color themes. This should make it easier to implement button texts.

### CODE REVIEW

File: react/Button/Readme.md [hunk 1]
```diff
@@ -13,12 +13,17 @@ when `<ButtonLink>` has:
...
-  <p><Button theme='danger' label='erase or destroy' /></p>
-  <p><Button theme='highlight' label='events communication' /></p>
-  <p><Button theme='secondary' label='cancel or second option' /></p>
+  {themes.map(theme => 
+    <p>{
+      props.map(
+        props => <Button label={theme} theme={theme} {...props} />
+      )
+    }</p>
+  )}
```
Comment (ptbrowne) [inline react/Button/Readme.md]: Here I display all themes with disabled and busy to easily debug variants.

File: react/ButtonAction/Readme.md [hunk 2]
```diff
@@ -5,14 +5,22 @@ Default type is `normal`
...
+    <ButtonAction label='Normal' rightIcon='openwith' disabled />
+  </p>
+  <p>
+    <ButtonAction type='new' leftIcon="plus" label='New' />
+    <ButtonAction type='new' leftIcon="plus" label='New' disabled />
+  </p>
+  <p>
+    <ButtonAction type='error' label='Error' rightIcon='file-none' />
+    <ButtonAction type='error' label='Error' rightIcon='file-none' disabled />
+  </p>
```
Comment (ptbrowne) [inline react/ButtonAction/Readme.md]: Added disabled variants to easily catch mistakes

File: stylus/components/button.styl [hunk 3]
```diff
@@ -404,40 +352,35 @@ $actionbtn--compact
...
-    border-color yourPink
-    background-color chablis
-    color pomegranate
+    actionbtnVariant(chablis, pomegranate, yourPink)
 
-    [data-action='icon']
-        border-color yourPink
+$actionbtn--new
+    actionbtnVariant(zircon, dodgerBlue, frenchPass)
+    border-width .0625rem // specific
```
Comment (GoOz) [inline stylus/components/button.styl]: Isn't that obviously specific? do we need the comment?

File: react/Button/Readme.md (hunk 1 above)
Comment (GoOz) [reply react/Button/Readme.md]: It appears that `secondary` & `danger-outline` when busy are broken then.

File: stylus/components/button.styl (hunk 3 above)
Comment (ptbrowne) [reply stylus/components/button.styl]: True, I removed the comment

File: react/Button/Readme.md (hunk 1 above)
Comment (ptbrowne) [reply react/Button/Readme.md]: Yes they were broken before. I fixed that in https://github.com/cozy/cozy-ui/pull/444/commits/7009bd436f6a7087655650e6e8886dea62ba9187

File: stylus/components/button.styl [hunk 4]
```diff
@@ -82,6 +82,9 @@ alpha = { name: 'alpha', bgColor: transparent, bdColor: white, focusColor: scien
 secondary = { name: 'secondary', bgColor: white, bdColor: silver, focusColor: silver, txtColor: black}
 danger-outline = { name: 'danger-outline', bgColor: white, bdColor: yourPink, focusColor: yourPink, txtColor: pomegranate}
 
+// We turn stylint off since without semicolons, stylus fails to
+// parse the following code block.
+// @stylint off
 for theme in highlight danger alpha secondary danger-outline
     $button--{theme['name']}
         background-color: theme['bgColor']
```
Comment (gregorylegarec) [inline stylus/components/button.styl]: I ❤️ semicolons.

File: stylus/components/button.styl [hunk 5]
```diff
@@ -76,99 +76,43 @@ $button
...
-    svg
-        fill currentColor
-
-    &[disabled]
-    &[aria-disabled=true]
+highlight = { name: 'highlight', bgColor: emerald, bdColor: emerald, focusColor: malachite, txtColor: white}
+danger = { name: 'danger', bgColor: pomegranate, bdColor: pomegranate, focusColor: monza, txtColor: white}
+alpha = { name: 'alpha', bgColor: transparent, bdColor: white, focusColor: scienceBlue, txtColor: white}
+secondary = { name: 'secondary', bgColor: white, bdColor: silver, focusColor: silver, txtColor: black}
+danger-outline = { name: 'danger-outline', bgColor: white, bdColor: yourPink, focusColor: yourPink, txtColor: pomegranate}
```
Comment (gregorylegarec) [inline stylus/components/button.styl]: As we aim to be able to easy build ButtonText  (or _no-outline_ buttons) from themes, and knowing that in those buttons color are "inverted" what do you think about naming attributes something like:
```styl
highlight  = { name: 'highlight', mainColor: emerald, secondaryColor: emerald, activeColor: malachite, contrastColor: white }
```

**Comment by gregorylegarec** [comment]
Looks great and elegant 👍

File: stylus/components/button.styl [hunk 6]
```diff
@@ -76,34 +76,59 @@ $button
...
+}
+alphaTheme = {
+    bgColor: transparent, bdColor: white, focusColor: scienceBlue, txtColor: white
+}
+secondaryTheme = {
+    bgColor: white, bdColor: silver, focusColor: silver, txtColor: black
+}
+dangerOutlineTheme = {
+    bgColor: white, bdColor: yourPink, focusColor: yourPink, txtColor: pomegranate
+}
```
Comment (gregorylegarec) [inline stylus/components/button.styl]: (copy pasted from an outdated comment)

As we aim to be able to easy build ButtonText  (or _no-outline_ buttons) from themes, and knowing that in those buttons color are "inverted" what do you think about naming attributes something like:
```styl
highlight  = { name: 'highlight', mainColor: emerald, secondaryColor: emerald, activeColor: malachite, contrastColor: white }
```

Comment (GoOz) [reply stylus/components/button.styl]: I kinda agree with @gregorylegarec on this. Could be a good time to do so.

Comment (ptbrowne) [reply stylus/components/button.styl]: done, kudos for the names !

**Pull request merged**